packages/react-ai/src/components/AIConversation/views/Controls/PromptControl.tsx

```diff
@@ -111,6 +112,14 @@ export const PromptControl: PromptControl = () => (
   </Container>
 );
 
+export const AutoHidablePromptControl = (): JSX.Element | undefined => {
+  const messages = React.useContext(MessagesContext);
+
+  if (!messages || messages.length === 0) {
+    return <PromptControl />;
+  }
+};
+
```

Comment:
@esauerbo I wasn't really sure if this was the best way to do this, but I figured internally our SuggestedPromptControl would show or hide automatically based on whether messages exist or not.

But if a customer were to compose it themselves they might find that automated behavior strange, so wanted to separate the auto hiding feature to it's own control. What do you think?

Reply:
I thinks that makes sense! 

Just spitballing here but one consideration is that there are a few other UI aspects (like showing the Avatar, disabling the `send` button, input placeholder text) that depend on the messages "state". Would we want to do the same thing in those cases? would it make sense to keep track of any messages metadata like # of messages in the context, or just perform that check within each component?

Reply:
Yeah I think it might make sense to expand the messages context to include that type of metadata if we find it helpful? Seems like a bit of a case by case scenario but I can see that making sense for showing/hiding the avatar. Not sure about `send` button or placeholder text.